airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -73,191 +73,129 @@
-        runMandatoryActivity((input) -> connectionDeletionActivity.deleteConnection(input), connectionDeletionInput);
-        return;
-      } else if (workflowState.isCancelled() || workflowState.isCancelledForReset()) {
-        runMandatoryActivity((input) -> jobCreationAndStatusUpdateActivity.jobCancelled(input),
-            new JobCancelledInput(
-                maybeJobId.get(),
-                maybeAttemptId.get(),
-                FailureHelper.failureSummaryForCancellation(maybeJobId.get(), maybeAttemptId.get(), failures, partialSuccess)));
-        resetNewConnectionInput(connectionUpdaterInput);
-      } else if (workflowState.isFailed()) {
-        reportFailure(connectionUpdaterInput);
-      } else {
-        // report success
-        reportSuccess(connectionUpdaterInput);
+      if (workflowState.isCancelled()) {
+        reportCancelledAndContinueWith(false, connectionUpdaterInput);
       }
-      continueAsNew(connectionUpdaterInput);
+
     } catch (final Exception e) {
       log.error("The connection update workflow has failed, will create a new attempt.", e);
-
-      reportFailure(connectionUpdaterInput);
-      continueAsNew(connectionUpdaterInput);
+      reportFailure(connectionUpdaterInput, null);
+      prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
     }
   }
 
-  private void reportSuccess(final ConnectionUpdaterInput connectionUpdaterInput) {
+  private void generateSyncWorkflowRunnable(ConnectionUpdaterInput connectionUpdaterInput) {
+    scopedForCancellationSyncWorkflow = Workflow.newCancellationScope(() -> {
+      connectionId = connectionUpdaterInput.getConnectionId();
+
+      if (connectionUpdaterInput.getWorkflowState() != null) {
+        workflowState = connectionUpdaterInput.getWorkflowState();
+      }
+
+      workflowState.setResetConnection(connectionUpdaterInput.isResetConnection());
+
+      Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
+
+      Workflow.await(timeToWait,
+          () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+
+      val needToProcessSync = !workflowState.isUpdated() && !workflowState.isDeleted();
+
+      if (needToProcessSync) {
+        jobId = getOrCreateJobId(connectionUpdaterInput);
+
+        attemptId = createAttemptId(jobId);
+
+        final GeneratedJobInput jobInputs = getJobInput();
+
+        reportJobStarting();
+        StandardSyncOutput standardSyncOutput = null;
+        try {
+          standardSyncOutput = runChildWorkflow(jobInputs);
+
+          checkifFailedAndSetFailStatus(standardSyncOutput);
+
+          if (workflowState.isFailed()) {
+            reportFailure(connectionUpdaterInput, standardSyncOutput);
+            prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          } else {
+            reportSuccess(connectionUpdaterInput, standardSyncOutput);
+            prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          }
```

Comment:
```suggestion
          if (workflowState.isFailed()) {
            reportFailure(connectionUpdaterInput, standardSyncOutput);
          } else {
            reportSuccess(connectionUpdaterInput, standardSyncOutput);
          }
          
          prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
```

nit: feels redundant to have duplicate behavior in each branch when the branch is only deciding if it should report.

Reply:
Davin add the same comment, the redundancy is on purpose. On the previous version the redundancy was avoided and we ended up with something like:
```
if (a) {
...
} else if (b) {
...
} else if (c) {
...
} else if (d) {
...
}

continueAsNew()
```

It was hard to see that the execution was returning in the a condition (temporal continue as new act as a return statement).

Having the redundancy is here to make the place where we return as explicit as possible.

Reply:
Although I like early exits in general, I feel the readability here is hampered by the brackets and it's easier to understand if we move `continueAsNew` out.

If we had something like this, I agree.
```
if (a) {
// do stuff
continueAsNew()
}
// do stuff
```

However,
```
if (a) {
...
continueAsNew()
} else if (b) {
...
continueAsNew()
} else if (c) {
... 
continueAsNew()
}
```
is more confusing as it makes me wonder why this is repeated.
```

Reply:
Done